lib/spack/spack/spec.py

```diff
@@ -426,7 +447,56 @@ def __init__(self, spec_like, *dep_like, **kwargs):
         # Spec(a, b) will copy a but just add b as a dep.
         for dep in dep_like:
             spec = dep if isinstance(dep, Spec) else Spec(dep)
-            self._add_dependency(spec)
+            self._add_dependency(spec, (spack.DEP_BUILD, spack.DEP_INSTALL))
+
+
+    def get_dependency(self, name):
```

Comment:
why not call this `get_depspec` and return the dep spec?  That probably scales better to any future edge properties we want to add.

The unpacking is nice now, and you could preserve it by making `DependencySpec` have an `__iter__` or `__getitem__` method, but using unpacking won't work if DependencySpec grows more things...